README - Programmatic Usage (#101)
hCaptcha/react-hcaptcha: README.md

</FormComponent>
```

#### Programmatic Usage

> In the event you want to call the hCaptcha client API directly you can do so by using the hook `useRef` and waiting for `onLoad` to be called. By waiting for `onLoad` the hCaptcha API will be ready and the hCaptcha client will have been setup. See the following example:

Reviewer: ```suggestion
In the event you want to call the hCaptcha client API directly, you can do so by using the hook `useRef` and waiting for `onLoad` to be called. By waiting for `onLoad` the hCaptcha API will be ready and the hCaptcha client will have been setup. See the following example:
```

```js
import { useEffect, useRef, useState } from "react";
import HCaptcha from "@hcaptcha/react-hcaptcha";

export default function Form() {
  const [token, setToken] = useState(null);
  const captchaRef = useRef(null);

  const onLoad = () => {

>     // this reaches out to the hcaptcha library and runs the

Reviewer: ```suggestion
    // this reaches out to the hCaptcha JS API and runs the
```

>     // execute function on it. you can use other functions as well

Reviewer: ```suggestion
    // execute function on it. you can use other functions as
```

>     // documented in the api:

Reviewer: ```suggestion
    // documented here:
```
